industrial_ci/src/workspace.sh

```diff
@@ -285,13 +290,19 @@ function ici_prepare_sourcespace {
             ici_guard rm -r "${sourcespace:?}/$file"
             ;;
         -*)
-            local file="${source:1}"
-            if [ ! -e "${sourcespace:?}/$file" ]; then
-              file="$(basename "$basepath")/$file"
-            fi
+            for file in "${source:1}" "$last/${source:1}" "$(basename "$basepath")/${source:1}"; do
+                if [ -e "${sourcespace:?}/$file" ]; then break; fi
+            done
             ici_log "Removing '${sourcespace:?}/$file'"
             ici_guard rm -r "${sourcespace:?}/$file"
             ;;
+        +*)
+            for file in "${source:1}" "$last/${source:1}"; do
+                if [ -e "${sourcespace:?}/$file" ]; then break; fi
+            done
+            ici_log "Moving '${sourcespace:?}/$file' to '${sourcespace:?}'"
+            ici_guard mv "${sourcespace:?}/$file" "${sourcespace:?}"
```

Comment:
This does not copy parent folders..

Reply:
Which parent folder you are referring to? I'm assuming that you would move folders below `$sourcespace` only.

Reply:
`repo/sub1/sub2/abc` would end up in `${sourcespace}/abc`

Reply:
Yes. Does that harm?
Of course, your syntax proposal that filters folders within a repo parent folder, will be much better. No risk for conflicts ;-)